fb.c

```diff
@@ -145,3 +145,148 @@ void destroy_framebuffer(const char *proc_fname)
     pr_debug("Destroying framebuffer %s\n", proc_fname);
     remove_proc_entry(proc_fname, NULL);
 }
+
+static int vcam_fb_check_var(struct fb_var_screeninfo *var,
+                             struct fb_info *info)
+{
+    if (!var->xres)
+        var->xres = 1;
+    if (!var->yres)
+        var->yres = 1;
+    if (var->xres > var->xres_virtual)
+        var->xres_virtual = var->xres;
+    if (var->yres > var->yres_virtual)
+        var->yres_virtual = var->yres;
+
+    var->xres_virtual = var->xoffset + var->xres;
+    var->yres_virtual = var->yoffset + var->yres;
+
+    var->bits_per_pixel = 24;
+
+    switch (var->bits_per_pixel) {
+    case 24:
+        var->red.offset = 0;
+        var->red.length = 8;
+        var->green.offset = 8;
+        var->green.length = 8;
+        var->blue.offset = 16;
+        var->blue.length = 8;
+        var->transp.offset = 0;
+        var->transp.length = 0;
+        break;
+    }
+    var->red.msb_right = 0;
+    var->green.msb_right = 0;
+    var->blue.msb_right = 0;
+    var->transp.msb_right = 0;
+
+    return 0;
+}
+
+static int vcam_fb_set_par(struct fb_info *info)
+{
+    switch (info->var.bits_per_pixel) {
+    case 24:
+        info->fix.visual = FB_VISUAL_TRUECOLOR;
+        break;
+    }
+    return 0;
+}
+
+static const struct fb_ops vcamfb_ops = {
+    .fb_set_par = vcam_fb_set_par,
+    .fb_check_var = vcam_fb_check_var,
+};
+
+static struct fb_fix_screeninfo vfb_fix = {
+    .id = "vcamfb",
+    .type = FB_TYPE_PACKED_PIXELS,
+    .visual = FB_VISUAL_TRUECOLOR,
+    .xpanstep = 1,
+    .ypanstep = 1,
+    .ywrapstep = 1,
+    .accel = FB_ACCEL_NONE,
+};
+
+static const struct fb_videomode vfb_default = {
+    .xres = 640,
```

Comment:
Can you share the default size with V4L2 part? That is, we only have to maintain one set of default resolution.